src/main/java/appeng/core/sync/packets/PacketClick.java

```diff
@@ -20,67 +20,56 @@
 
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+
 import appeng.api.AEApi;
 import appeng.api.definitions.IComparableDefinition;
 import appeng.api.definitions.IItems;
 import appeng.api.implementations.items.IMemoryCard;
 import appeng.api.implementations.items.MemoryCardMessages;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.items.tools.ToolNetworkTool;
 import appeng.items.tools.powered.ToolColorApplicator;
 
 
-public class PacketClick extends AppEngPacket
+public class PacketClick implements AppEngPacket, AppEngPacketHandler<PacketClick, AppEngPacket>
 {
 
-	final int x;
-	final int y;
-	final int z;
-	final int side;
-	final float hitX;
-	final float hitY;
-	final float hitZ;
+	private int x;
+	private int y;
+	private int z;
+	private int side;
+	private float hitX;
+	private float hitY;
+	private float hitZ;
 
 	// automatic.
-	public PacketClick( ByteBuf stream )
+	public PacketClick()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)